static/src/javascripts/projects/commercial/modules/messenger/scroll.ts

```diff
@@ -31,49 +37,62 @@ const domRectToRect = (rect) => ({
 	right: rect.right,
 });
 
-const sendCoordinates = (iframeId, domRect) => {
+const sendCoordinates = (iframeId: string, domRect: DOMRect) => {
 	iframes[iframeId].respond(null, domRectToRect(domRect));
 };
 
-const getDimensions = (id) => [id, iframes[id].node.getBoundingClientRect()];
+const getDimensions = (id: string): [string, DOMRect] => [
+	id,
+	iframes[id].node.getBoundingClientRect(),
+];
 
-const isIframeInViewport = function (item) {
+const isIframeInViewport = function (
+	this: Viewport,
+	item: [string, DOMRect],
+): boolean {
 	return item[1].bottom > 0 && item[1].top < this.height;
 };
 
-const onIntersect = (changes) => {
+const onIntersect: IntersectionObserverCallback = (changes) => {
 	visibleIframeIds = changes
 		.filter((_) => _.intersectionRatio > 0)
 		.map((_) => _.target.id);
 };
 
+// typescript complains about an async event handler, so wrap it in a non-async function
```

Comment:
Could we instead `void fastdom.measure` and return `void` if `taskQueued`?

Reply:
Ah yeah of course! 🤦‍♂️

Reply:
No need for the comment anymore
```suggestion
```